python/covid_web/covid_web/rest.py

```diff
@@ -79,7 +82,34 @@ def get_app(self):
         return self
 
     def get_available_predictions(self):
-        return jsonify(self.available_predictions)
+        result = []
+        for prediction in self.available_predictions:
+            prediction['link'] = url_for("covid19_get_specific_prediction", country=prediction['country'], prediction=prediction['prediction'])
+            result.append(prediction)
+
+        return jsonify(result)
+
+    def get_predictions_by_country(self, country: str):
+        result = []
+        for prediction in self.predictions:
+            if prediction["short_name"] == country:
+                result.append(prediction)
+
+        if len(result) == 0:
+            abort(404)
+
+        return jsonify(result)
+
+    def get_predictions_by_name(self, date: str):
+        result = []
+        for prediction in self.predictions:
+            if prediction["date_name"] == date:
+                result.append(prediction)
```

Comment:
We prefer list comprehensions for these
```suggestion
    result = [
        prediction
        for prediction in self.predictions:
        if prediction["date_name"] == date
    ]
```

Reply:
Changed.